### PR TITLE
Update ADDON_HOOKS.md links and code snippets

### PR DESCRIPTION
**Pull request by sweatypitts**
This addresses issue #5702. It updates links to the latest tag (`v2.4.3`), fixes broken links, adds links that should have been there but weren't, and updates code snippets to reflect current code.

### CODE REVIEW

File: ADDON_HOOKS.md [hunk 1]
```diff
@@ -255,7 +254,7 @@ Usually used to import assets into the application.
 ```js
 // https://github.com/yapplabs/ember-colpick/blob/master/index.js
 included: function colpick_included(app) {
-  this._super.included.apply(this, arguments);
+  this._super.included(app);
```
Comment (stefanpenner) [inline ADDON_HOOKS.md]: apply is required

Comment (sweatypitts) [inline ADDON_HOOKS.md]: Good catch. The source should probably get updated then: https://github.com/yapplabs/ember-colpick/blob/master/index.js#L10

I'll update the example here and submit a PR to ember-colpick.

**Comment by sweatypitts** [comment]
@stefanpenner Requested change has been made. I will make my way over to ember-colpick later today.

File: ADDON_HOOKS.md (hunk 1 above)
Comment (stefanpenner) [inline ADDON_HOOKS.md]: > Good catch. The source should probably get updated then: https://github.com/yapplabs/ember-colpick/blob/master/index.js#L10

:+1:

**Comment by stefanpenner** [comment]
:+1: from me, @rwjblue r?

File: ADDON_HOOKS.md [hunk 2]
```diff
@@ -150,12 +150,11 @@ Designed to manipulate requests in development mode.
...
 *Note:* that this should only be used in development, and if you need the same behavior in production you'll need to configure your server.
 
-
 **Examples:**
 
-- [ember-cli-content-security-policy](https://github.com/rwjblue/ember-cli-content-security-policy/blob/v0.3.0/index.js#L25)
+- [ember-cli-content-security-policy](https://github.com/rwjblue/ember-cli-content-security-policy/blob/v0.5.0/index.js#L84)
 
-- [history-support-addon](https://github.com/ember-cli/ember-cli/blob/v0.1.15/lib/tasks/server/middleware/history-support/index.js#L13)
+- [history-support-addon](https://github.com/ember-cli/ember-cli/blob/v2.4.3/lib/tasks/server/middleware/history-support/index.js#L13)
```
Comment (nathanhammond) [inline ADDON_HOOKS.md]: https://github.com/ember-cli/ember-cli/blob/v2.4.3/lib/tasks/server/middleware/history-support/index.js#L25

**Comment by nathanhammond** [comment]
:+1: with the one tweak. @sweatypitts update and we'll merge! Thank you very much!

**Comment by rwjblue** [comment]
LGTM

**Comment by sweatypitts** [comment]
K, made change noted by @nathanhammond. Thanks, guys.

**Comment by nathanhammond** [comment]
@homu r+

**Pull request merged**